.changeset/bright-donkeys-exist.md

```diff
@@ -0,0 +1,5 @@
+---
+"@primer/components": patch
+---
+
+Action Menu correctly fires onAction callback after close.
```

Comment:
Might want to update these release notes to reflect the fact that it will be called _before_ close now